Update zh_cn.json

linzexin022 opened the pull request:
Updated Chinese Simplified translation. By the way, I did not find the Chinese Translation in the latest version(1.16.3)?

lorilith commented on src/main/java/com/github/levoment/superaxes/TreeChopper.java:
```diff
@@ -69,113 +69,55 @@ public void traverseTree(World world, BlockPos pos) {
+        //
         // Loop while there are elements in the stack
         while (!stackOfBlockPositions.isEmpty()) {
-            // Set the blocks to look for
```
lots of redundant code here. It was condensed.

lorilith commented:
sadly, the recursion over large trees is too slow (such as the modded ones from terrestria). I will need to rework this for speed

Levoment commented:
Thanks for the comment and commit. Yes, if a leave is found on the algorithm it will scan for nearby leaves to make sure it covers all the logs of a tree. That causes it to also chop down trees that are connected by leaves. Removing that would also cause the removal of the part of harvesting leaves unless I find a way to only collect the leaves that belong to a single tree. I've been thinking of a way, but because trees are made by separate blocks and not a single entity, nothing has come to mind yet. I'm open to suggestions and help on that one. I also noticed that trees, like Acacia, that have diagonal logs, fail to get fully chopped down, but when I tried to scan for those diagonals the code grew too slow. Apart from that, there are two ways to have the harvest of big trees not lag and those ways are: by spreading the harvesting across ticks or by using a separate thread that sleeps every certain amount of milliseconds. But once again, when I tried to do that, rather than lagging for 1 second or 2 seconds when chopping down a tree, the lag would be less and barely noticeable, but the fps would drop by 20+ on big trees until everything was harvested. For that reason I decided to leave the current algorithm in which it would only lag for 1 or 2 seconds when chopping big trees rather than spreading the harvesting across ticks or time.

lorilith commented:
Lag seems to be due to leaf spread to multiple large trees. i am adding a 2d radius (defaulting to 5) to help prevent leaf spread and lag time. I am not sure that these are ideal as the leaf breaking may detach leaves from their mother tree if too close. 

I suppose it is a matter of "pick your poison". I may look into how the fast leaf decay mods deal with leaves.

Sadly, i can bypass the leaf checks if not breaking leaves (since a log shouldnt really be hanging out 2 blocks away from the rest of the logs) but if breaking leaves, the whole thing slows way down, requiring the distance check. To be fair, the original mod also has this issue; I experienced 5-30 second delay depending on number of trees. I reduced the delay to about 3 seconds with the range check (but also with the much larger search pattern of a 3x3 cube).

Levoment commented:
Hi @lorilith,

Can you re-open the pull-request. I managed to improve the code in the PR and would like to merge your changes into the project before applying my changes. :)

linzexin022 committed:
Update zh_cn.json

Updated Chinese Simplified translation. By the way, I did not find the Chinese Translation in the latest version(1.16.3)?

linzexin022 committed:
Update zh_cn.json

Levoment reviewed (approved):
Hi,

Thanks for the update. I will merge it and upload the version with the changes soon.
